akka-http-tests/src/test/scala/akka/http/server/directives/BasicDirectivesSpec.scala

```diff
@@ -26,4 +29,22 @@ class BasicDirectivesSpec extends RoutingSpec {
       } ~> check { responseAs[String] shouldEqual "GET" }
     }
   }
-}
\ No newline at end of file
+
+  "The `extractUri` directive" should {
+    "extract from the request url" in {
+      Get("https://example.com/foo") ~> {
+        extractUri { echoComplete }
+      } ~> check { responseAs[String] shouldEqual "https://example.com/foo" }
+    }
+    "extract from a X-Request-URI header" in {
+      Get() ~> addHeader(RawHeader("x-request-uri", "http://example.com/foo")) ~> {
+        extractUri { echoComplete }
+      } ~> check { responseAs[String] shouldEqual "http://example.com/" }
```

Comment:
akka-http doesn't know anything about the `X-Request-URI` header and just ignores it. So, there's not really much to see here. The directive itself isn't tested because it just return the `uri` field of the `HttpRequest`. How this uri field is composed is tested in lower levels (but maybe not sufficiently documented).